src/compiler/core.ts

```diff
@@ -1653,7 +1653,7 @@ namespace ts {
      */
     export function compose<T>(...args: ((t: T) => T)[]): (t: T) => T;
     export function compose<T>(a: (t: T) => T, b: (t: T) => T, c: (t: T) => T, d: (t: T) => T, e: (t: T) => T): (t: T) => T {
-        if (e) {
+        if (!!e) {

```

Comment:
This is a pretty good example of what TS would now consider suspicious-enough code to error on. Making `e` optional would also have fixed this.